crates/re_types_builder/src/codegen/python/mod.rs

```diff
@@ -1991,8 +1981,167 @@ return pa.UnionArray.from_buffers(
+                        format!("{variant_kind_list} += 1"),
+                    )
+                } else {
+                    let ignore_type_check = if type_based_variants {
+                        ""
+                    } else {
+                        // my-py doesn't know that this has the right type now.
+                        "# type: ignore[arg-type]"
+                    };
+                    (
+                        format!("value_offsets.append(len({variant_kind_list}))"),
+                        format!("{variant_kind_list}.append(value.inner) {ignore_type_check}"),
+                    )
+                };
+                variant_list_push_arms.push_indented(3, value_offset_update, 1);
+                variant_list_push_arms.push_indented(3, append_to_variant_kind_list, 1);
+                variant_list_push_arms.push_indented(3, format!("types.append({})", idx + 1), 1); // 0 is reserved for nulls
+
+                // Converting the variant list to a pa array.
+                let variant_list_to_pa_array = match &field.typ {
+                    Type::Object(fqname) => {
+                        let field_type_name = &objects[fqname].name;
+                        format!(
+                            "{field_type_name}Batch({variant_kind_list}).as_arrow_array().storage"
+                        )
+                    }
+                    Type::Unit => {
+                        format!("pa.nulls({variant_kind_list})")
+                    }
+                    Type::UInt8
+                    | Type::UInt16
+                    | Type::UInt32
+                    | Type::UInt64
+                    | Type::Int8
+                    | Type::Int16
+                    | Type::Int32
+                    | Type::Int64
+                    | Type::Bool
+                    | Type::Float16
+                    | Type::Float32
+                    | Type::Float64
+                    | Type::String => {
+                        let datatype = quote_arrow_datatype(&arrow_registry.get(&field.fqname));
+                        format!("pa.array({variant_kind_list}, type={datatype})")
+                    }
+                    Type::Array { .. } | Type::Vector { .. } => {
+                        return Err(format!(
+                            "We lack codegen for arrow-serialization of unions containing lists. Can't handle type {}",
+                            field.fqname
+                        ));
+                    }
+                };
+                child_list_push.push_indented(1, format!("{variant_list_to_pa_array},"), 1);
+            }
+
+            let batch_type_imports = quote_local_batch_type_imports(&obj.fields);
+            Ok(format!(
+                r##"
+{batch_type_imports}
+from typing import cast
+
+# Ensure data is iterable.
+try:
+    iter(data) # type: ignore[arg-type]
+except TypeError:
+    data = [data] # type: ignore[list-item]
```

Comment:
This only works if none of the element-types themselves are iterable. Specifically I think this will cause a problem in the situation of a Union that has a string as a union-arm.

Reply:
Hmmm true. This is surprisingly tricky. For starters I can check for the union itself and any of its arms, but we might also want to check for all type aliases of each arm 😱

Reply:
how about [4cd630d](https://github.com/rerun-io/rerun/pull/6258/commits/4cd630d2a1765f4d06ff7be7b00d45a591e7f1e1)